crates/astria-sequencer/src/asset/state_ext.rs

```diff
@@ -72,3 +72,162 @@ pub(crate) trait StateWriteExt: StateWrite {
 }
 
 impl<T: StateWrite> StateWriteExt for T {}
+
+#[cfg(test)]
+mod test {
+    use astria_core::sequencer::v1::asset::{
+        Denom,
+        Id,
+    };
+    use cnidarium::StateDelta;
+
+    use super::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    };
+
+    #[tokio::test]
+    async fn get_ibc_asset_non_existent() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let state = StateDelta::new(snapshot);
+
+        let asset = Id::from_denom("asset_0");
+
+        // gets for non existing assets fail
+        state
+            .get_ibc_asset(asset)
+            .await
+            .expect_err("gets for non existing ibc assets should fail");
+    }
+
+    #[tokio::test]
+    async fn has_ibc_asset() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        let asset_string = "asset_0";
+        let asset = Id::from_denom(asset_string);
+        let denom = Denom::from_base_denom(asset_string);
+
+        // non existing calls are ok for 'has'
+        assert!(
+            !state
+                .has_ibc_asset(asset)
+                .await
+                .expect("'has' for non existing ibc assets should be ok"),
+            "query for non existing asset should return false"
+        );
+
+        state
+            .put_ibc_asset(asset, &denom)
+            .expect("putting ibc asset should not fail");
+
+        // existing calls are ok for 'has'
+        assert!(
+            state
+                .has_ibc_asset(asset)
+                .await
+                .expect("'has' for existing ibc assets should be ok"),
+            "query for existing asset should return true"
+        );
+    }
+
+    #[tokio::test]
+    async fn put_ibc_asset_simple() {
+        let storage = cnidarium::TempStorage::new().await.unwrap();
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // can write new
+        let asset_string = "asset_0";
+        let asset = Id::from_denom(asset_string);
+        let denom = Denom::from_base_denom(asset_string);
```

Comment:
```suggestion
        let denom = Denom::from_base_denom(asset_string);
        let id = denom.id()
```

you can remove the `let asset = Id::from_denom(asset_string);` and just use `denom.id()` for the `Id` parameters, same for other tests